File: src/main/java/jukebot/commands/Volume.java
package jukebot.commands;

import jukebot.audioutilities.AudioHandler;
import jukebot.utils.Command;
import jukebot.utils.CommandProperties;
import jukebot.utils.Context;
import jukebot.utils.Helpers;

@CommandProperties(aliases = {"vol"}, description = "Adjust the player volume", category = CommandProperties.category.CONTROLS)
public class Volume implements Command {

    private final String brick = "\u25AC";
    private final int maxBricks = 10;

    public void execute(final Context context) {

        final AudioHandler player = context.getAudioPlayer();

        if (!player.isPlaying()) {
            context.embed("Not Playing", "Nothing is currently playing.");
            return;
        }

        if (context.getArgString().isEmpty()) {
            final int vol = player.player.getVolume();
            context.embed("Player Volume", calculateBricks(vol) + " `" + vol + "`");
            return;
        }

        if (!context.isDJ(false)) {
            context.embed("Not a DJ", "You need to be a DJ to use this command.\n[See here on how to become a DJ](https://jukebot.xyz/faq)");
            return;
        }

        player.player.setVolume(Math.min(Helpers.Companion.parseNumber(context.getArgString(), 100), 200));

        final int vol = player.player.getVolume();
        context.embed("Player Volume", calculateBricks(vol) + " `" + vol + "`");

    }

    private String calculateBricks(int volume) {
        final float percent = (float) volume / 200;
        final int blocks = (int) Math.floor(maxBricks * percent);

        final StringBuilder sb = new StringBuilder("[");

        for (int i = 0; i < maxBricks; i++) {
            if (i == blocks) {
                sb.append("](http://jukebot.xyz)");
            }

            sb.append(brick);
        }

        if (blocks == maxBricks) {
            sb.append("](http://jukebot.xyz)");
        }

        return sb.toString();
    }
}

File: src/main/java/jukebot/commands/Volume.kt
package jukebot.commands

import jukebot.framework.Command
import jukebot.framework.CommandCategory
import jukebot.framework.CommandProperties
import jukebot.framework.Context
import jukebot.utils.Helpers
import kotlin.math.min

@CommandProperties(aliases = ["vol", "v"], description = "Adjust the player volume", category = CommandCategory.CONTROLS)
class Volume : Command(ExecutionType.REQUIRE_MUTUAL) {

    override fun execute(context: Context) {
        val player = context.getAudioPlayer()

        if (!player.isPlaying) {
            return context.embed("Not Playing", "Nothing is currently playing.")
        }

        // If listeners == 1, ask them to use Discord volume slider?

        if (context.args.isEmpty()) {
            val vol = player.player.volume
            return context.embed("Player Volume", "${Helpers.createBar(vol, 250, 10)} `$vol`")
        }

        if (!context.isDJ(false)) {
            return context.embed("Not a DJ", "You need to be a DJ to use this command.\n[See here on how to become a DJ](https://jukebot.serux.pro/faq)")
        }

        player.player.volume = min(context.args.first().toIntOrNull() ?: 100, 250)

        val vol = player.player.volume
        context.embed("Player Volume", "${Helpers.createBar(vol, 250, 10)} `$vol`")
    }


}
